extractor/extractor.py

```diff
@@ -0,0 +1,223 @@
+import bs4
+from bs4 import Comment
+import re
+from urllib.request import urlopen
+import PIL.Image
+import requests
+from pprint import pprint
+import contents
+
+req_session = ""
+
+def parse(url):
+    global req_session
+
+    req_session = requests.Session()
+    file = req_session.get(url).text    # To request Html from URL
+
+    # Remove the comment to read local files for testing purpose
+    # file = open('./test_html.html','r').read()
+
+    soup = bs4.BeautifulSoup(file,'lxml')
+    soup = clean_soup(soup)
+
+    extract(soup)
+
+
+def extract(soup):
+    contents_list = contents.Contents()
+
+    # Extract data from head
+    head = soup.find('head')
+    extract_title_tag(head, contents_list)
+
+    # Extract data from body
+    body_soup = soup.find('body')
+    iterate_over_dom(body_soup, contents_list)
+
+    ''' 
+    Stores the extracted content
+    Refer to file contents.py to see the class structure 
+    '''
+    contents_list.print_contents()
+
+
+def is_content(node):
+    flag = 0
+    _content = {}
+
+    # check if content is text
+    isText, text = is_text(node)
+    if isText == 1:
+        _content = contents._Content()
+        _content.contentType = 'Text'
+        _content.text = text
+        return 1, _content
+
+    # check if content is image
+    isImage, image = is_image(node)
+    if isImage == 1:
+        _content = contents._Content()
+        _content.contentType = 'Image'
+        _content.image = image
+        return 1, _content
+
+    # check if content is video
+    isVideo, video = is_video(node)
+    if isVideo == 1:
+        _content = contents._Content()
+        _content.contentType = 'Video'
+        _content.video = video
+        return 1, _content
+
+    # check if content is quote
+    # check if content is tweet
+    # check if content is instagram post
+    # check if content is pinterest pin
+    # check if content is youtube video
```

Comment:
I guess you can TODO for each item

Reply:
Added collective TODO for all items for now.